pkg/metrics/timer.go

```diff
@@ -0,0 +1,42 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Timer is the struct that keeps track of each of the sections.
+type Timer struct {
+	startTime    time.Time
+	totalTime    time.Time
+	listOfStages []string
+	listOfTimes  []time.Duration
```

Comment:
We shouldn't rely on the index of listOfTimes to correspond to listOfStages. For one, this will break down if we want to use this in a different command, such as wait-for or destroy.

Reply:
Was worried about the order in which the error messages were printed and hence used a list.